### PR TITLE
#186427584 : apply gzip in createEventsBatchAsync

### PR DESCRIPTION
**Pull request by XinyuLuan**
- [x] Unit test

### CODE REVIEW

File: src/main/java/com/moesif/api/IAPIController.java [hunk 1]
```diff
@@ -46,7 +47,18 @@ void createEventAsync(final EventModel body,
...
-                                final APICallBack<HttpResponse> callBack) throws JsonProcessingException;
+                                final APICallBack<HttpResponse> callBack) throws IOException;
+
+
+    /**
+     * Add multiple API Events in a single batch
+     * @param    body    The events to create
+     * @param    callBack Called after the HTTP response is received
+     * @param    useGzip Option to convert body to gzip type
+     * @throws JsonProcessingException on error creating event
```
Comment (praves77) [inline src/main/java/com/moesif/api/IAPIController.java:58]: is it still throws `JsonProcessingException` ? function signature shows it is `IOException`

File: src/main/java/com/moesif/api/IAPIController.java [hunk 2]
```diff
@@ -46,7 +47,18 @@ void createEventAsync(final EventModel body,
...
+     */
+    void createEventsBatchAsync(final List<EventModel> body,
+                                final APICallBack<HttpResponse> callBack, boolean useGzip) throws IOException;
```
Comment (praves77) [inline src/main/java/com/moesif/api/IAPIController.java:61]: third parameter `useGZip` can be on next line to keep it consistent.

File: src/main/java/com/moesif/api/IAPIController.java (hunk 1 above)
Comment (XinyuLuan) [reply src/main/java/com/moesif/api/IAPIController.java:58]: Good catch, changed

File: src/main/java/com/moesif/api/IAPIController.java (hunk 2 above)
Comment (XinyuLuan) [reply src/main/java/com/moesif/api/IAPIController.java:61]: It's in next line in intellij

File: src/main/java/com/moesif/api/controllers/APIController.java [hunk 3]
```diff
@@ -119,22 +120,34 @@ public Map<String, String> createEventsBatch(
...
 
         QueryInfo qInfo = getQueryInfo("/v1/events/batch");
-
-        //prepare and invoke the API call request to fetch the response
         final HttpRequest _request = getClientInstance().postBody(qInfo._queryUrl, qInfo._headers, APIHelper.serialize(body));
 
+        if(useGzip){
+            executeRequestAsync(_request, callBack, true);
+        }
+
```
Comment (praves77) [inline src/main/java/com/moesif/api/controllers/APIController.java:150]: Replace all such calls with one liner.
```
executeRequestAsync(_request, callBack, useGzip);
```

Comment (XinyuLuan) [reply src/main/java/com/moesif/api/controllers/APIController.java:150]: changed

**Review by praves77 — APPROVED** [review]

**Pull request merged**